### PR TITLE
Task #216485: fixed issue= when we click on not mark on calendar then not able to see status of clicked dates.

### PR DESCRIPTION
**Pull request by upendraTekdi**
## Description
[Explain the purpose and context of this pull request. What feature does it implement or what issue does it resolve?]

### Type of change

Please choose appropriate options.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
bugFixed

## Screenshots/GIFs (if applicable)
[Include any screenshots or GIFs demonstrating the changes made, especially for UI changes.]

### How Has This Been Tested?

Please describe the tests that you ran to verify your changes in the below checkboxes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Ran Test A
- [ ] Ran Test B

### Checklist:
- [x] I have Added Required translations
- [ ] I have tagged appropriate issue ID to PR
- [ ] I have User theme variables for theming or introduced in theme.
- [ ] I have fixed Lint issues
- [ ] Project not creating any kind of build issues.
- [ ] My changes generate no new warnings
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

## Summary by CodeRabbit

- **New Features**
	- Added support for a new attendance status "Not marked" to improve clarity in attendance records.
	- Enhanced Class Attendance History page with formatted date ranges and dynamic user ID fetching for personalized attendance data.
	- Updated Dashboard to refine the display of cohorts and attendance information based on improved data handling.
	- MyClassDetails page now checks for both cohort ID and user ID before fetching data, ensuring more accurate cohort details are displayed.

- **Bug Fixes**
	- Fixed syntax issues in the Class Attendance History page that affected the initialization of marked dates.
	- Addressed missing commas in array declarations across various components, improving code stability.

- **Refactor**
	- Refactored data handling in MyClassDetails for better clarity and efficiency in displaying class details.
	- Adjusted the cohortList function in CohortServices to accept an additional "name" parameter, enhancing the flexibility of cohort data retrieval.

- **Documentation**
	- Updated interfaces to include an optional "name" field, aligning with changes in data retrieval methods.

### CODE REVIEW

File: src/components/AttendanceStatus.tsx [hunk 1]
```diff
@@ -34,6 +34,7 @@ function AttendanceStatus({ status, onUpdate }: AttendanceStatusProps) {
       message = 'Half-day';
       break;
     case 'notmarked':
+    case 'Not marked':
```
Comment (itsvick) [inline src/components/AttendanceStatus.tsx:37]: @upendraTekdi why there are two case for same functionality, is enum not set for the same?

File: src/components/AttendanceStatus.tsx [hunk 2]
```diff
@@ -34,6 +34,7 @@ function AttendanceStatus({ status, onUpdate }: AttendanceStatusProps) {
...
       message = 'Attendance not marked';
       break;
     case 'Future date':
```
Comment (itsvick) [inline src/components/AttendanceStatus.tsx:40]: @upendraTekdi make sure we not use spaces status

File: src/components/AttendanceStatus.tsx (hunk 1 above)
Comment (upendraTekdi) [reply src/components/AttendanceStatus.tsx:37]: may be first we are using notmarked as per api response , but now in api response we are getting it as  Not marked so thats why we are using both in that case

File: src/components/AttendanceStatus.tsx (hunk 2 above)
Comment (upendraTekdi) [reply src/components/AttendanceStatus.tsx:40]: yes.. used case is right as per api response

File: src/components/AttendanceStatus.tsx (hunk 1 above)
Comment (itsvick) [reply src/components/AttendanceStatus.tsx:37]: Get it confirmed from the backend team and keep one which works

Comment (upendraTekdi) [reply src/components/AttendanceStatus.tsx:37]: okay sure

**Pull request merged**